src/doc/rustdoc/src/how-to-write-documentation.md

```diff
@@ -0,0 +1,78 @@
+# How to write documentation
+
+This chapter intends to explain how to not only write documentation but also on
+how to write **good** ones. But something to keep in mind when writing
+documentation: you write as much for others as for you so it's important to be
+clear! The more the better. If an item is public, then it should be documented.
+
+## Basic structure
+
+If possible, each item's documentation should try to follow this structure:
+
+```text
+[short sentence explaining what it is]
+
+[more detailed explanation]
+
+[at least one code example that users can copy/paste to try it]
+
+[even more advanced explanations if necessary]
+```
+
+Pretty simple but important to follow. The code example is really important:
+it might help your users to have a better understanding of what an item is
+used for and/or how to use it.
```

Comment:
I think it's risky to describe documentation as 'simple' to produce.  Perhaps:

```markdown
This basic structure should be straightforward to follow when writing your
documentation and, while you might think that a code example is trivial,
the examples are really important because they can help your users to
understand what an item is, how it is used, and for what purpose it exists.
```